include/glow/Graph/Graph.h

```diff
@@ -261,10 +270,78 @@ class Function final : public Named {
 
   const Module *getParent() const { return parent_; }
 
+  /// Logs the node creation with a list of input nodes.
+  void logNodeCreation(Node *newNode, llvm::ArrayRef<Node *> inputs) {
+    if (dumpCompilationLogOpt) {
+      std::string logStr = "";
+      logStr.append("[FULL SCOPE: ");
+      logStr.append(logCtx_.getFullScopeName());
+      logStr.append("]");
+      logStr.append(" --- CREATE { (");
+      logStr.append(newNode->getKindName());
```

Comment:
I think it might be useful to prefix these values with the name of the field (i.e. the output should be `Kind: Transpose` instead of just `Transpose`). In your sample output, some nodes are named the same as their kind name and it's hard to tell which is which.

Reply:
Great suggestion!